keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,252 @@
+or set to null, the existing behavior is maintained for backward compatibility. 
+This means that Workload objects will not be deleted by Kueue, 
+aligning with the behavior before the introduction of this feature.
+
+### Risks and Mitigations
+
+- **R**: In clusters with a large number of existing finished and deactivated Workloads 
+(thousands, tens of thousands, etc.), it may take a significant amount of time
+to delete all Workloads that qualify for deletion. Since the 
+reconciliation loop is effectively synchronous, this may impact the reconciliation 
+of new jobs. From the user's perspective, it may seem like Kueue's initialization 
+time is very long for that initial run until all expired objects are deleted.\
+**M**: Unfortunately, there is no easy way to directly mitigate this without complicating the code.
+A potential improvement would be to have a dedicated internal queue that handles deletion 
+outside the reconciliation loop, or to delete expired objects in batches so that after 
+a batch of deletions is completed, any remaining expired objects would be skipped until 
+the next run of the loop. However, this would go against the synchronous nature of the reconciliation loop.
+Another currently available option would be to limit client qps and burst when the feature is enabled,
+allowing to mitigate the burst kube-apiserver load.
+
+## Design Details
+
+### API
+```go
+// Configuration is the Schema for the kueueconfigurations API
+type Configuration struct {
+    // …
+    // ObjectRetentionPolicies provides configuration options for automatic deletion
+    // of Kueue-managed objects. A nil value disables all automatic deletions.
+    // +optional
+    ObjectRetentionPolicies *ObjectRetentionPolicies `json:"objectRetentionPolicies,omitempty"`
+}
+
+// ObjectRetentionPolicies holds retention settings for different object types.
+type ObjectRetentionPolicies struct {
+    // Workloads configures retention for finished Workloads.
+    // A nil value disables automatic deletion of finished Workloads.
+    // +optional
+    Workloads *WorkloadRetentionPolicy `json:"workloads,omitempty"`
+}
+
+// WorkloadRetentionPolicy defines when finished Workloads should be deleted.
+type WorkloadRetentionPolicy struct {
+    // AfterFinished is the duration to wait after a Workload finishes
+    // before deleting it.
+    // A duration of 0 will delete immediately.
+    // A nil value disables automatic deletion.
+    // Represented using metav1.Duration (e.g. "10m", "1h30m").
+    // +optional
+    AfterFinished *metav1.Duration `json:"afterFinished,omitempty"`
+    // AfterDeactivated is the duration to wait after a Workload has been deactivated
+    // before deleting it.
+    // A duration of 0 will delete immediately.
+    // A nil value disables automatic deletion.
+    // Represented using metav1.Duration (e.g. "10m", "1h30m").
+    // +optional
+    AfterDeactivated *metav1.Duration `json:"afterDeactivated,omitempty"`
```

Comment:
We need to describe and warn users this may actually delete objects not created by Kueue directly.

Reply:
Described deletion of jobsets and how that might delete objects not created by Kueue

Reply:
This is not just about JobSet, in general any  workload objects (like Job or JobSet) which are manged by Kueue.

Reply:
Have changed the description, this one should reflect the functionality better

Reply:
sgtm